roles/v2/pool/src/error.rs

```diff
@@ -7,6 +7,7 @@ use std::{
 #[derive(std::fmt::Debug)]
 pub enum PoolError {
     Io(std::io::Error),
+    InvalidExtranonce(String),
```

Comment:
is this variant needed somewhere? here (https://github.com/stratum-mining/stratum/pull/513/files#diff-a2101e207a5d63e663b5e05e4fb90216cc3973ef8027c011c0a58417464cb73b) I added an error code relating to this to return in an `OpenMiningChannelError` which may cover the case you had in mind